include/gridtools/stencil/frontend/run.hpp

```diff
@@ -102,17 +179,38 @@ namespace gridtools {
             constexpr empty_spec<core::forward> execute_forward() { return {}; }
             constexpr empty_spec<core::backward> execute_backward() { return {}; }
 
-            template <class... Msses>
-            constexpr spec<Msses...> multi_pass(spec<Msses>...) {
+            template <class Mss, class... Msses>
+            constexpr spec<Mss, Msses...> multi_pass(spec<Mss>, spec<Msses>...) {
                 return {};
             }
 
+            template <class... Ts>
+            void multi_pass(Ts...) {
+                static_assert(sizeof...(Ts) < 0, "Unexpected arguments of gridtools::stencil::multi_pass.");
+            }
+
             template <size_t>
             struct arg {};
 
+            template <class Interval, class F>
+            struct functor_validator {
+                static_assert(core::is_valid_functor<F, Interval>::value, "Invalid elementary functor");
```

Comment:
Is this supposed to be a message to the user? Then we have to be more precise. If not, we used to use the `GT_INTERNAL_ERROR_MESSAGE`. How did you use it recently?

Reply:
question applies to many of the new static asserts in this file

Reply:
`GT_INTERNAL_ERROR_MESSAGE` is a bad idea here. This assert triggers on user provided type.

Reply:
done. PTAL